src/boat_simulator/boat_simulator/nodes/mock_data/mock_data_node.py

```diff
@@ -0,0 +1,131 @@
+import random
+import sys
+
+import rclpy
+from custom_interfaces.msg import DesiredHeading, HelperHeading, SailCmd
+from rclpy.node import Node
+
+import boat_simulator.common.constants as Constants
+
+# based on following:
+# https://docs.ros.org
+# /en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Py-Publisher-And-Subscriber.html
+# the purpose of this node is to publish to all topics that there is
+# subscribers in physics engine node so send goal code can work.
+
+
+class MockDataNode(Node):
+
+    def __init__(self):
+        super().__init__("mock_data")
+        self.__declare_ros_parameters()
+
+        self.desired_heading_pub = self.create_publisher(
+            msg_type=DesiredHeading,
+            topic=Constants.PHYSICS_ENGINE_SUBSCRIPTIONS.DESIRED_HEADING,
+            qos_profile=self.qos_depth,
+        )
+        self.sail_trim_tab_angle_pub = self.create_publisher(
+            msg_type=SailCmd,
+            topic=Constants.PHYSICS_ENGINE_SUBSCRIPTIONS.SAIL_TRIM_TAB_ANGLE,
+            qos_profile=self.qos_depth,
+        )
+
+        timer_period = self.pub_period  # seconds
+        self.timer = self.create_timer(timer_period, self.timer_callback)
+
+    def __declare_ros_parameters(self):
+        """Declares ROS parameters from the global configuration file that will be used in this
+        node. This node will monitor for any changes to these parameters during execution and will
+        update itself accordingly.
+        """
+        self.get_logger().debug("Declaring ROS parameters...")
+        self.declare_parameters(
+            namespace="",
+            parameters=[
+                ("pub_period_sec", rclpy.Parameter.Type.DOUBLE),
+                ("mock_sail_trim_tab", rclpy.Parameter.Type.BOOL),
+                ("mock_desired_heading", rclpy.Parameter.Type.BOOL),
+                ("qos_depth", rclpy.Parameter.Type.INTEGER),
+            ],
+        )
+
+        all_parameters = self._parameters
+        for name, parameter in all_parameters.items():
+            value_str = str(parameter.value)
+            self.get_logger().debug(f"Got parameter {name} with value {value_str}")
+
+    def timer_callback(self):
+        if self.mock_desired_heading:
+            self.publish_mock_desired_heading()
+        if self.mock_sail_trim_tab:
+            self.publish_mock_sail_trim_tab_angle()
+
+    def publish_mock_desired_heading(self):
+        """Publishes mock wind sensor data."""
+        heading = random.uniform(-179.99, 180.0)
```

Comment:
Could you please make the interval configurable through `globals.yaml` as well. This would make it easier to adjust on the fly and help avoid the need to rebuild the boat simulator package every time it's changed.